autopr/agents/brain_agent/simple_v1.py

```diff
@@ -0,0 +1,44 @@
+from typing import Union
+
+from git.repo import Repo
+
+from autopr.models.artifacts import Issue
+from .base import BrainAgentBase
+
+import structlog
+
+from ...models.events import EventUnion
+
+log = structlog.get_logger()
+
+
+class BasicBrainAgent(BrainAgentBase):
+    def _generate_pr(
+        self,
+        event: EventUnion,
+    ) -> None:
+        issue = event.issue
+
+        # Switch to the base branch
+        self.commit_service.overwrite_new_branch()
```

Comment:
i've only just clanged at this so far, but the one recommendation I have is that it might be helpful to add one more layer of abstraction inside the BrainAgentBase API. The current approach still requires that the user keep track of which service/agent is responsible for which step of the process. e.g.

```python
class BrainAgentBase:
    ...
    def plan_pull_request(...):
        return self.pull_request_agent.plan_pull_request(...)
```

This might make it easier to e.g. make a change isolated to a single step of the process by overriding a single function in the brain agent without modifying the overall logic for the sequence of steps taken to generate the PR.

Reply:
I was thinking we shouldn't enforce "steps of the process" in the brain agent, a new implementation should be able to change them entirely, or invoke the subagents autonomously. In that sense, a brain agent implementation doesn't need to use the subagents (pr planning and codegen), but is welcome to.

If you'd like to change the pull request planning process, you can implement a new pull request agent, and specify it in your config. Or do you see some other benefit for only adjusting how the pull request agent is invoked by the brain agent?

Reply:
maybe it might help if you demonstrate some modified agents so it's clearer how/where to make changes to achieve a given outcome, and what kinds of flexibility the new design supports

Reply:
To be honest, I'm not sure if it makes sense to have flexibility on this level, or if the "plan => codegen" architecture is optimal. But it will allow us to experiment with new approaches easily. After I hit a couple of the other TODOs on the roadmap I'll have a go at another brain agent architecture.